### DISCUSSION TITLE
TypeError: Cannot read properties of undefined (reading 'tap')

### DISCUSSION THREAD

**Discussion by pushkargupta9807** [web-infra-dev/rspack#3424] [General]
![image](https://github.com/web-infra-dev/rspack/assets/93791633/e42dd564-439a-460c-9638-3dc32a14fd3c)

Hello, 
I started getting this error when I upgraded to the latest v 0.2.0 . 

Any pointers/ideas how to get around this issue? It seems that webpack is installed as @rspack/dev-server depends on it. In general whenever I upgrade to newer versions, I start getting errors which take days to debug and resolve. Would really appreciate any help here. Thanks

**Comment by hardfist** [comment]
its weird we should not call this  webpack plugin in rspack，maybe you can help investigate why this hook is registered in rspack

**Reply by pushkargupta9807** [reply]
I am fairly new to this domain - can you please elaborate on what you think might be the problem so that I can dig deeper into it .

**Reply by pushkargupta9807** [reply]
@hardfist : I have a hunch that this plugin is being called because of the web-worker format being defaulted to 'array-push'
https://webpack.js.org/configuration/output/#outputchunkformat

And I am using the worker syntax as below:
![image](https://github.com/web-infra-dev/rspack/assets/93791633/d5392027-3644-4251-a492-17fbb63e526b)

Thoughts?

**Reply by Venryx** [reply]
I had the same error, and found it was caused by my having this line in one of my js files:
```
const ORM_EncoderWorker = require("worker-loader!opus-media-recorder/encoderWorker.umd.js");
```

So basically, I was trying to use the "worker-loader" extension, despite it not being compatible with rspack (apparently). Maybe there's a way to make them work together, but I just commented out the line, since I'm not using the worker atm anyway.

 Btw, I found the cause by adding the line below to the webpack js file that errored:
```
	apply(compiler) {
		throw new Error("This shouldn't get called!");             <-- I added this line to find what was registering the "compilation" callback below
		compiler.hooks.thisCompilation.tap(
			"ArrayPushCallbackChunkFormatPlugin",
			compilation => {
				compilation.hooks.additionalChunkRuntimeRequirements.tap(           <-- this line errored
```

**Reply by hardfist** [reply]
since rspack supports web-worker out of box, you don't need worker-loader to support it, and we have no plan to support worker-loader in the near future
